table/block_based_table_reader.cc

```diff
@@ -2061,7 +2061,11 @@ Status BlockBasedTable::DumpTable(WritableFile* out_file) {
 }
 
 void BlockBasedTable::Close() {
-  rep_->filter_entry.Release(rep_->table_options.block_cache.get());
+  const bool force_erase = true;
+  // The filter_entry could have a pointer to the table and must be deleted
+  // before the table is closed
+  rep_->filter_entry.Release(rep_->table_options.block_cache.get(),
+                             force_erase);
```

Comment:
Although its comments indicates the intention of erasing the filter, its code does not do that. The key is formed based on "rep_->footer.metaindex_handle()" while the filter inserted with a key that is formed with "rep_->filter_handle".

The force_erase approach is more efficient and also cleaner way of doing this:
1) In LRUCache it does not require a second hash lookup to find the key and erase it.
2) It does not require re-constructing the key in two places in the code, which could results into bugs like this.

If you confirm that intention of  line 2079 to 2081 were to erase index and filter, I can go ahead deleting those lines, and having index and filter erased with force_erase option.